core/sql/sqlcomp/PrivMgrDesc.cpp

```diff
@@ -550,14 +554,28 @@ void PrivMgrCoreDesc::interpretChanges( const bool before,       // in
 // ----------------------------------------------------------------------------
 bool PrivMgrDesc::limitToGrantable( const PrivMgrDesc& other )
 {
-
   bool result = false;
-  PrivMgrCoreDesc logicalTablePrivs(other.tableLevel_);
-  // TDB - include column level privileges
-
- if ( tableLevel_.limitToGrantable(logicalTablePrivs) )
-    result = TRUE;
+  if ( tableLevel_.limitToGrantable(other.tableLevel_) )
+    result = true;
 
-   return result;
+  // Consider table level privileges when checking column privs
+  PrivMgrCoreDesc temp = other.tableLevel_;
+  for (int i = 0; i < columnLevel_.entries(); i++)
+  {
+    // Find associated column in other Desc list
+    int index = other.getColumnPriv(columnLevel_[i].getColumnOrdinal());
+    if (index >= 0)
+    {
+      temp.unionOfPrivs(other.columnLevel_[index]);
+      if (columnLevel_[i].limitToGrantable(temp))
+        result = true;
+    }
+    else
+    {
+      if (columnLevel_[i].limitToGrantable(temp))
```

Comment:
Did you intend the "if (columnLevel_[i].limitToGrantable(temp)) etc." to be the same in both the "if" and "else" code paths? (Just curious why you bothered with the "else".)